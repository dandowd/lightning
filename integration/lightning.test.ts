import { spawn } from 'child_process'

describe('it should read from stdin', () => {
  let sut: ReturnType<typeof spawn>

  beforeEach(() => {
    sut = spawn('ts-node', ['../src/index.ts'], { cwd: __dirname })
  })

  afterEach(() => {
    // Ensure that the process is killed after done() is called
    sut.kill()
  })

  it('should read from stdin', (done) => {
    sut.stdout?.on('data', data => {
      expect(data.toString()).toBe('log: hello')
      done()
    })

    sut.stdin?.setDefaultEncoding('utf-8')
    sut.stdout?.pipe(process.stdout)

    sut.stdin?.write('log: hello')
  })
})
