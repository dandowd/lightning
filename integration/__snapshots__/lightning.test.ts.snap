// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should read from stdin should write struck area to stdout 1`] = `
[
  "lightning alert for 86315:Fahey Brooks",
  "lightning alert for 035:McClure Ford",
  "lightning alert for 55708:Denis Meadows",
  "lightning alert for 132:Raynor Avenue",
  "lightning alert for 02115:Mayer Park",
  "lightning alert for 7052:Arely Parkway",
  "lightning alert for 5645:Block Corners",
  "lightning alert for 178:Buddy Burgs",
  "lightning alert for 2818:Hand Inlet",
  "lightning alert for 512:Suellen Fall",
  "lightning alert for 565:Hauck Station",
  "lightning alert for 325:Sunshine Wall",
  "lightning alert for 3781:Ardyce Lights",
  "lightning alert for 015:Dach Village",
  "lightning alert for 3685:Nanie Loaf",
  "lightning alert for 89860:Volkman Highway",
  "lightning alert for 06591:Brady Loaf",
  "lightning alert for 03575:O'Connell Cape",
  "lightning alert for 90430:Gutkowski Mount",
  "lightning alert for 06149:Novella Islands",
  "lightning alert for 04755:Abernathy Walks",
  "lightning alert for 4709:Stark Dam",
  "lightning alert for 22521:Evalyn Tunnel",
  "lightning alert for 015:Schulist Lodge",
  "lightning alert for 3319:Jaida Glen",
  "lightning alert for 67873:Heaven Curve",
  "lightning alert for 5130:Marvin Cove",
  "lightning alert for 8549:Mante Oval",
  "lightning alert for 6664:Wolf Locks",
  "lightning alert for 94908:Boyle Cape",
  "lightning alert for 2043:Tatyana Run",
  "lightning alert for 70395:Guilford Squares",
  "lightning alert for 3996:Wolff Canyon",
  "lightning alert for 47332:Inger Drive",
  "lightning alert for 79323:Jess Inlet",
  "lightning alert for 084:Langworth Orchard",
  "lightning alert for 083:Elenore Squares",
  "lightning alert for 8691:Marcela Centers",
  "lightning alert for 9656:Jorja Ramp",
  "lightning alert for 51081:Ayden Lights",
  "lightning alert for 23837:Cortney Causeway",
  "lightning alert for 9100:Hirthe Stream",
  "lightning alert for 9974:Christian Glens",
  "lightning alert for 41396:Steuber Creek",
  "lightning alert for 6983:Darby Lodge",
  "lightning alert for 0668:Lukas Square",
  "lightning alert for 37771:Neha Shores",
  "lightning alert for 95392:Thiel Meadows",
  "lightning alert for 491:Donn Trafficway",
  "lightning alert for 8212:Denesik Fall",
  "lightning alert for 8652:Cronin Isle",
  "lightning alert for 5992:Kierra Place",
  "lightning alert for 4830:Schmeler Ports",
  "lightning alert for 3719:Dach Course",
  "lightning alert for 87042:Little Lights",
  "lightning alert for 07724:Harlow Ridge",
  "lightning alert for 360:Hilpert Shores",
  "lightning alert for 647:Harber Skyway",
  "lightning alert for 55860:Gleason Route",
  "lightning alert for 261:Hudson Land",
  "lightning alert for 259:Syed Pines",
  "lightning alert for 346:Sariah Trafficway",
  "lightning alert for 758:Rogahn Trafficway",
  "lightning alert for 59421:Toy Center",
  "lightning alert for 723:Grimes Extensions",
  "lightning alert for 2620:Howell Dale",
  "lightning alert for 514:Sylvan Shore",
  "lightning alert for 6580:Kozey Trafficway",
  "lightning alert for 3133:Quigley Orchard",
  "lightning alert for 2835:Aspen Rapid",
  "lightning alert for 2721:Dezzie Radial",
  "lightning alert for 65723:Schiller Place",
  "lightning alert for 972:Reichert Gateway",
  "lightning alert for 108:Fumiko Plaza",
  "lightning alert for 8123:Tasia Springs",
  "lightning alert for 92843:Smitham Summit",
  "lightning alert for 77807:Tula Summit",
  "lightning alert for 2714:Cephus Oval",
  "lightning alert for 662:Willodean Branch",
  "lightning alert for 61001:Talan Cliff",
  "lightning alert for 459:Hauck Lakes",
  "lightning alert for 706:Bernie Corner",
  "lightning alert for 05742:Roxanne Ports",
  "lightning alert for 234:Sarahi Islands",
  "lightning alert for 76858:Hoeger Plains",
  "lightning alert for 52731:Cyntha Fords",
  "lightning alert for 61057:Cummerata Summit",
  "lightning alert for 62110:Frankie Mountains",
  "lightning alert for 811:Alfredo Motorway",
  "lightning alert for 917:Constance Land",
  "lightning alert for 215:Kertzmann Meadow",
  "lightning alert for 941:Bridget Port",
  "lightning alert for 6403:Esther Valley",
  "lightning alert for 98209:McKenzie Spurs",
  "lightning alert for 530:Hermiston Fork",
  "lightning alert for 74082:Renaldo River",
  "lightning alert for 69242:Micah Lodge",
  "lightning alert for 674:Rogahn Inlet",
  "lightning alert for 3432:Schaden Station",
  "lightning alert for 88167:Bernier Camp",
  "lightning alert for 55380:Baker Cliff",
  "lightning alert for 21771:Huel Drives",
  "lightning alert for 47952:Simeon Stravenue",
  "lightning alert for 4626:Augusta Freeway",
  "lightning alert for 46201:Erick Lights",
  "lightning alert for 855:Julian Walks",
  "lightning alert for 118:Mitchell Corners",
  "lightning alert for 7100:Luvinia Union",
  "lightning alert for 3593:White Estates",
  "lightning alert for 5166:Amerigo Grove",
  "lightning alert for 11865:Halvorson Port",
  "lightning alert for 605:White Trace",
  "lightning alert for 555:Macey Drive",
  "lightning alert for 992:Dibbert Village",
  "lightning alert for 6849:King Run",
  "lightning alert for 5320:Dee Plains",
  "lightning alert for 66868:Sharif Key",
  "lightning alert for 226:Erie Lake",
  "lightning alert for 4444:Neal Village",
  "lightning alert for 96072:Willms Streets",
  "lightning alert for 572:Cremin Lights",
  "lightning alert for 0179:Klein Mission",
  "lightning alert for 269:Welch Tunnel",
  "lightning alert for 8412:Bogan Corners",
  "lightning alert for 46360:Nikolai Fords",
  "lightning alert for 704:Braun Circle",
  "lightning alert for 27359:Elease Estates",
  "lightning alert for 2448:Jakob Way",
  "lightning alert for 141:Quitzon Union",
  "lightning alert for 363:Dare Isle",
  "lightning alert for 178:Jane Underpass",
  "lightning alert for 3002:Rodriguez Motorway",
  "lightning alert for 152:Eliga Loop",
  "lightning alert for 84393:Larson Isle",
  "lightning alert for 621:Arnav Fields",
  "lightning alert for 746:Hyatt Trail",
  "lightning alert for 313:Cruickshank Forge",
  "lightning alert for 585:Effertz Center",
  "lightning alert for 11270:Blick Island",
  "lightning alert for 4344:Doreen Ridges",
  "lightning alert for 6167:Reilly Hills",
  "lightning alert for 48707:Jenny Freeway",
  "lightning alert for 003:Glenda Circle",
  "lightning alert for 5014:Edie Lodge",
  "lightning alert for 59276:Stefanie Street",
  "lightning alert for 9896:Milburn Loop",
  "lightning alert for 31010:Beer Freeway",
]
`;
